### PR TITLE
Fixes for Display handling

### PR DESCRIPTION
**Pull request by JamiePhonic**
This is my attempt at updates to display.py and client.py to fix issues with displays > 128px wide and causing divide by 0 errors on squeezelite-ESP32 devices using LED-VU display without a display connected.

### CODE REVIEW

**Comment by marcelveldt** [comment]
please leave out the dependency bumps, we use a separate process for that.

File: aioslimproto/display.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def __init__(
         self.width = width
         self.visualisation_type = visualisation_type
         # NOTE: height can only be 32, its not adjustable
-        self.image = Image.new("1", (width, 32))
+        self.image = Image.new("1", (self.width, 32))
```
Comment (marcelveldt) [inline aioslimproto/display.py:115]:
```suggestion
        self.image = Image.new("1", (width, 32))
```

prefer the local var (as you already have it)

**Comment by JamiePhonic** [comment]
Accepted your suggestion to prefer the local var and rolled back project.toml to rever dependancy bumps

File: aioslimproto/client.py [hunk 2]
```diff
@@ -966,6 +977,20 @@ def _process_setd(self, data: bytes) -> None:
             if display_height:
                 self.display_control.height = display_height
 
+            # Trigger an event callback for "PLAYER_DISPLAY_RESOLUTION"
+            resolution = f"{display_width} x {display_height}"
+            self.callback(self, EventType.PLAYER_DISPLAY_RESOLUTION, resolution)
```
Comment (marcelveldt) [inline aioslimproto/client.py:982]: should the callback maybe be called after setting these variables below ?

**Comment by marcelveldt** [comment]
Sorry for the late review, next time feel free to ping me sooner as I sometimes can not overview the many repositories I'm involved with.

PR Looks good, just one minor comment and CI seems to be failing on the linter/tests

File: aioslimproto/client.py (hunk 2 above)
Comment (JamiePhonic) [reply aioslimproto/client.py:982]: do you mean after the check for display.width or display.height == 0 ?

**Comment by JamiePhonic** [comment]
Moved the callback to the end of the function as suggested

**Comment by marcelveldt** [comment]
Linter still fails. 
Please install the tests config in your venv with pip install -e .[test]
and then run pre-commit install
perform a full lint on the branch with pre-commit run --all-files

**Comment by JamiePhonic** [comment]
> Linter still fails. Please install the tests config in your venv with pip install -e .[test] and then run pre-commit install perform a full lint on the branch with pre-commit run --all-files

I did that, it looks like the lint failures are coming from Ruff itself?
```
Warning:  hook id `ruff-check` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
Warning:  hook id `ruff-format` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
Warning:  hook id `check-executables-have-shebangs` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
Warning:  hook id `end-of-file-fixer` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
Warning:  hook id `trailing-whitespace` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
```

**Comment by JamiePhonic** [comment]
Here's what i got:
```
@JamiePhonic ➜ /workspaces/aioslimproto (main) $ pre-commit run --all-files
[WARNING] hook id `ruff-check` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
[WARNING] hook id `ruff-format` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
[WARNING] hook id `check-executables-have-shebangs` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
[WARNING] hook id `end-of-file-fixer` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
[WARNING] hook id `trailing-whitespace` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
🐶 Ruff Linter...........................................................Passed
🐶 Ruff Formatter........................................................Passed
🐍 Check Python AST......................................................Passed
🔠 Check for case conflicts..............................................Passed
ℹ️  Check docstring is first.............................................Passed
🧐 Check that executables have shebangs..................................Passed
｛ Check JSON files..................................(no files to check)Skipped
💥 Check for merge conflicts.............................................Passed
🔗 Check for broken symlinks.........................(no files to check)Skipped
✅ Check TOML files......................................................Passed
✅ Check code for common misspellings....................................Passed
🕵️  Detect Private Keys..................................................Passed
⮐  Fix End of Files......................................................Passed
🛑 Don't commit to main branch...........................................Failed
- hook id: no-commit-to-branch
- exit code: 1
✄  Trim Trailing Whitespace..............................................Passed
@JamiePhonic ➜ /workspaces/aioslimproto (main) $ 
```

**Comment by marcelveldt** [comment]
> That's what i get for using Windows i guess 🤷‍♂️

Haha, yeah Windows is not your best platform to develop python - I gave up on that long, long time ago.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -9,14 +9,14 @@ repos:
         types: [python]
         entry: scripts/run-in-env.sh ruff check --fix
         require_serial: true
-        stages: [commit, push, manual]
+        stages: [pre-commit, pre-push, manual]
```
Comment (marcelveldt) [inline .pre-commit-config.yaml:12]: this change is unrelated to the PR and under normal circumstances shouldnt be needed - can you tell me the motivation to change it ?

**Review by marcelveldt — APPROVED** [review]
Nice, thanks!
